terraform/transform_provider.go

```diff
@@ -605,6 +605,43 @@ func (t *ProviderConfigTransformer) transformSingle(g *Graph, c *configs.Config)
 		t.proxiable[key] = !diags.HasErrors()
 	}
 
+	if mod.ProviderRequirements != nil {
```

Comment:
Is this covered in a test, and if not, can it be? (I know the answer to that isn't always straightforward).

Reply:
Yes, there's a context test added which covers this, and I'll be adding more as I re-validate the overall behavior within the terraform package itself. This PR was getting big, and I wanted to get the parser and validation in ASAP.